src/Extension/FluentFilteredExtension.php

```diff
@@ -76,6 +76,16 @@ public function augmentSQL(SQLSelect $query, DataQuery $dataQuery = null)
                 $locale->ID,
             ]
         );
+
+        $joinAlias = $this->getLocalisedTable($table, $locale->Locale);
+
+        // Only select records that either have their own Localisation, or have been added as a Filtered Locale.
```

Comment:
Note, this also fails if using FluentFilteredExtension but not FluentExtension. ;)